features/TESTS/filesystem/fat_file_system/main.cpp

```diff
@@ -29,10 +29,27 @@ using namespace utest::v1;
 HeapBlockDevice bd(128*BLOCK_SIZE, BLOCK_SIZE);
 
 
+/*
 void test_format() {
     int err = FATFileSystem::format(&bd);
     TEST_ASSERT_EQUAL(0, err);
 }
+*/
+
+void test_format() {
+    int err = -1;
+    FATFileSystem fs("");
+
+    err = fs.mount(&bd, false);
+    TEST_ASSERT_EQUAL(0, err);
+
+    err = fs.format(&bd);
+    TEST_ASSERT_EQUAL(0, err);
+
+    err = fs.unmount();
+    TEST_ASSERT_EQUAL(0, err);
+}
```

Comment:
I think this may be binding to tightly to the ChanFS implementation.

It seems unintuitive that a filesystem requires being mounted before being formatted, usually the opposite is true. In most filesystems, formatting is a standalone operation that occurs before mounting. Is there a reason formatting should be dependent on mounting?